flamedisx/source.py

```diff
@@ -31,6 +31,12 @@ class Source:
     # Final observable dimensions; for use in domain / cross-domain
     final_dimensions = tuple()
 
+    # Avoid variable stepping over these inner_dimensions
+    no_step_dimensions = tuple()
+
+    # Any non-hidden variable extra_dimensions
+    bonus_dimensions = tuple()
```

Comment:
@robertsjames  Should this be `extra_dimensions` now? I see that you renamed `bonus_dimensions` to `extra_dimensions` in the block source codes (or is `bonus_dimensions` the general name, and `extra_dimensions` a specific use of it??)

Reply:
`extra_dimensions` is now the generic name for any additional dimensions you want a block to see. If they represent an internally contracted hidden variable, they will get added to `inner_dimensions` (this will be the case for the ions in the NEST implementation). Otherwise they get added to `bonus_dimensions`, where you can do what you like with them (this is the case here for quanta_produced_noStep).